### PR TITLE
Added support for MKRZero and Created Board Test Suite

### PR DESCRIPTION
**Pull request by ShawnHymel**
Arduino MKRZero pins and board definitions added for ports/atmel-samd/boards. Created a set of tests for similar ports. These tests can be run individually to ensure that pins and buses work properly when creating a new port.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -60,13 +60,13 @@ Other
    supported using the `Adafruit CircuitPython SD
    library <https://github.com/adafruit/Adafruit_CircuitPython_SD>`__)
 -  `Arduino Zero <https://www.arduino.cc/en/Main/ArduinoBoardZero>`__
+-  `Arduino MKR Zero <https://store.arduino.cc/arduino-mkrzero>`
 
 "Third-party" or "non-Adafruit" boards
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 -  `Electronic Cats Meow Meow <https://electroniccats.com/gomeow/>`__
--  `Electronic Cats CatWAN USB Stick <https://electroniccats.com/producto/catwan_usb_stick/>`__
```
Comment (tannewt) [inline README.rst]: The trailing __ is needed for RST links here and above. You can see the output on GitHub by clicking View File: https://github.com/adafruit/circuitpython/blob/f8ded46ad69ebba2de5b791bde16c50c4d2b1d1b/README.rst

**Comment by dhalbert** [comment]
This error happened in the doc build in travis, I think because the link is a redirect (redirects to `https://store.arduino.cc/usa/arduino-mkrzero` or presumably something else that's location-dependent).
```
Warning, treated as error:
/home/travis/build/adafruit/circuitpython/README.rst:63:'any' reference target not found: https://store.arduino.cc/arduino-mkrzero
The command "(! var_search "${TRAVIS_TESTS-}" docs || sphinx-build -E -W -b html . _build/html)" exited with 2.
```

File: README.rst (hunk 1 above)
Comment (ShawnHymel) [reply README.rst]: Will do. I'll cancel the PR and re-submit with the test code in the other repo. Thanks!

**Comment by ShawnHymel** [comment]
@dhalbert 

Thanks for catching that. I'm new to Travis and RST, so this is all learning for me :)

**Comment by ShawnHymel** [comment]
Closing PR. Will submit MKRZero port separately and move test suite to different repo as per @tannewt 's request.

**Pull request closed without merging**